pkg/apk/repo.go

```diff
@@ -246,7 +246,7 @@ func (p *PkgResolver) GetPackagesWithDependencies(packages []string) (toInstall
 			return nil, nil, fmt.Errorf("could not find package %s", pkgName)
 		}
 		// do not add it to toInstall, as we want to have it in the correct order with dependencies
-		dependenciesMap[pkgName] = pkgs[0]
+		dependenciesMap[pkgs[0].Name] = pkgs[0]
```

Comment:
I see this is already like this, but why do we only care about the 0th result of ResolvePackage?

Reply:
Because that's the highest matching one. ResolvePackage returns all of the packages that match in descending priority.